Add promotion feature(again)

Jacob-Pilon opened the pull request.

Jacob-Pilon committed:
Add promotion feature

mwhit176 reviewed (changes requested):
Pawn promotion is not working for me on this branch

I notice that a lot of your code is in the `move()` method. This method is not aptly named - this is the function responsible for populating the possible moves, **not** responsible for actually moving the pieces. Please make sure that you're testing your changes locally before making a PR and let me know if you're having trouble getting the program running properly.

Jacob-Pilon committed:
Merge branch 'main' into CA-8

Jacob-Pilon committed:
Added for visibility

Jacob-Pilon committed:
Fixes for promotion

Jacob-Pilon committed:
Final changes for promotion

Had a pain in the butt trying to figure out how push my changes (I do not use eclipse). Let me know if it still doesnt work I got it working on my machine will figure out if I missed any files.

mwhit176 commented on Chess/src/pieces/Black_Bishop.png:
I'm not a fan of having all of these new images when they already exist elsewhere in the project. Can you please remove these from the PR?

mwhit176 reviewed (approved):
Everything looks pretty good besides the extra images which should not be included in this merge.

**Nitpicks:**
![image](https://github.com/user-attachments/assets/219aad9e-2961-4ead-87eb-d221875de362)
When promoting a pawn, it appears in both squares at the same time while the dialogue shows.

Additionally, one can currently simply close out of the dialogue to choose not to promote a pawn - a player should be forced to promote.
